__tests__/endpoints.test.js

```diff
@@ -227,13 +244,56 @@ describe("/api/users", () => {
       .then(({ body: { users } }) => {
         expect(users).toHaveLength(4);
         users.forEach((user) => {
-          expect(user).toHaveProperty("username");
-          expect(user).toHaveProperty("name");
-          expect(user).toHaveProperty("avatar_url");
           expect(typeof user.username).toBe("string");
           expect(typeof user.name).toBe("string");
           expect(typeof user.avatar_url).toBe("string");
         });
       });
   });
 });
+
+describe("/api/articles?", () => {
+  test("GET 200 - Responds with a list of articles with a topic value of cats", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        articles.forEach((article) => {
+          expect(article.topic).toEqual("cats");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("author");
+          expect(article).toHaveProperty("body");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("article_img_url");
+        });
+      });
+  });
+  test("GET 200 - Responds with a list if all articles if no query given after query character", () => {
+    // Would this case still expect a 200 status code? Or a 400?
+    return request(app)
+      .get("/api/articles?")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("author");
+          expect(article).toHaveProperty("title");
+          expect(article).toHaveProperty("article_id");
+          expect(article).toHaveProperty("topic");
+          expect(article).toHaveProperty("created_at");
+          expect(article).toHaveProperty("votes");
+          expect(article).toHaveProperty("article_img_url");
+          expect(article).toHaveProperty("comment_count");
+        });
+      });
+  });
+  test("GET 404 - Invalid query value given. No article with that topic", () => {
+    return request(app)
+      .get("/api/articles?topic=nonExistent")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Article not found");
+      });
+  });
```

Comment:
what if the topic exists but there are no articles associated with it? eg "paper"